src/plonk/eval.rs

```diff
@@ -150,6 +161,14 @@ pub trait Eval<F: PrimeField> {
             .try_fold(F::ZERO, |acc, value| Ok(acc + value?))
     }
 }
+impl<F: PrimeField, T: GetEvaluateData<F>> Eval<F> for T {}
+
+pub trait EvalGrouped<F: PrimeField>: GetEvaluateData<F> {
+    fn eval_grouped(&self, _poly: &GroupedPoly<F>) -> impl Iterator<Item = Result<Vec<F>, Error>> {
```

Comment:
as commented above, `eval_grouped` is just evaluation of a vector of Expressions. We don't need a separate method here.

Reply:
Well that's how I keep your original design, where the `eval` method takes `Multipoly`, which is what you were flipping `Expression` with. We definitely don't need to take vector of expression as a way to pass expressions grouped by degree, it's bad design when we already have a wrapper over them.